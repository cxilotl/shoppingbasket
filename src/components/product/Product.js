import React from 'react';
import PropTypes from 'prop-types';
import cssStyles from "./Product.module.scss";

const Product = ({ name, onIncrementing, onDecrementing }) => {
  return (
    <div className={ cssStyles.product }>
      <h4>{ name }</h4>
      <div className={ cssStyles.controls }>
        <button type="button" name="product-increment-btn" onClick={ onIncrementing }>+</button>
        <button type="button" name="product-decrement-btn" onClick={ onDecrementing }>-</button>
      </div>
    </div>
  );
};

Product.propTypes = {
  name: PropTypes.string,
  onIncrementing: PropTypes.func,
  onDecrementing: PropTypes.func
};

export default Product;