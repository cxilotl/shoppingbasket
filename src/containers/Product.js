import { connect } from 'react-redux';
import Product from "../components/product/Product";
import { addProduct, removeProduct } from '../services/cartActions';

const mapStateToProps = null;

const mapDispatchToProps = (dispatch) => {
  return {
    addProduct: (product) => dispatch(addProduct(product)),
    removeProduct: (product) => dispatch(removeProduct(product))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Product);