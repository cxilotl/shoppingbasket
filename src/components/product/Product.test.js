import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import Product from './Product';

describe('Product Item', () => {
  const productName = 'Beans';

  it('Snapshot', () => {
    const component = renderer.create(
      <Product
        name={ productName }
        onIncrementing={ jest.fn() }
        onDecrementing={ jest.fn() }
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('When interacting with the product item, the component', () => {
    let handleOnAdd;
    let handleOnRemove;

    beforeEach(() => {
      handleOnAdd = jest.fn();
      handleOnRemove = jest.fn();
    });

    it('should increment itself to the shopping cart', () => {
      const { getByText } = render (
        <Product
          name={ productName }
          onIncrementing={ handleOnAdd }
          onDecrementing={ handleOnRemove }
        />
      );
      fireEvent.click(getByText('+'));
      expect(handleOnAdd).toHaveBeenCalled();
    });

    it('should decrement itself to the shopping cart', () => {
      const { getByText } = render (
        <Product
          name={ productName }
          onIncrementing={ handleOnAdd }
          onDecrementing={ handleOnRemove }
        />
      );
      fireEvent.click(getByText('-'));
      expect(handleOnRemove).toHaveBeenCalled();
    });
  });

});