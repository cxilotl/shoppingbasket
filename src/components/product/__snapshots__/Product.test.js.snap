// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Item Snapshot 1`] = `
<div
  className="product"
>
  <h4>
    Beans
  </h4>
  <div
    className="controls"
  >
    <button
      name="product-increment-btn"
      onClick={[MockFunction]}
      type="button"
    >
      +
    </button>
    <button
      name="product-decrement-btn"
      onClick={[MockFunction]}
      type="button"
    >
      -
    </button>
  </div>
</div>
`;
